### PR TITLE
Code Scanning API support for retrying

### PR DESCRIPTION
**Pull request by GeekMasher**
### Dependency Updates:
* [`Pipfile`](diffhunk://#diff-230078d672f10d17463a8a6265cad825b790885898256a3365be90685caac58dL9-R9): Updated `ghastoolkit` dependency version from `0.13.1` to `0.13.4`.

### Retry Logic Enhancements:
* [`ghascompliance/__main__.py`](diffhunk://#diff-24fc4335ffe34eec4fbe4bfe1334676cae163cf01b414dd9982708c910ee052fR53-R54): Added `--retry-count` and `--retry-sleep` arguments to configure the number of retries and the sleep duration between retries.
* [`ghascompliance/__main__.py`](diffhunk://#diff-24fc4335ffe34eec4fbe4bfe1334676cae163cf01b414dd9982708c910ee052fR191-R192): Passed the new `retry_count` and `retry_sleep` arguments to the `ComplianceChecker` class.
* [`ghascompliance/checks.py`](diffhunk://#diff-4ab62e7bfe032a0d9e697e396311aa75967bcdd00cb6e845be67f146526f6011R34-R35): Updated the `__init__` method to accept `retry_count` and `retry_sleep` parameters and initialize them. [[1]](diffhunk://#diff-4ab62e7bfe032a0d9e697e396311aa75967bcdd00cb6e845be67f146526f6011R34-R35) [[2]](diffhunk://#diff-4ab62e7bfe032a0d9e697e396311aa75967bcdd00cb6e845be67f146526f6011R44-R46)
* [`ghascompliance/checks.py`](diffhunk://#diff-4ab62e7bfe032a0d9e697e396311aa75967bcdd00cb6e845be67f146526f6011L80-R87): Modified the `checkCodeScanning` method to use the retry parameters for code scanning and added debug logging for retries.

### CODE REVIEW

File: ghascompliance/__main__.py [hunk 1]
```diff
@@ -50,6 +50,8 @@
     "--github-policy-path",
     default=os.path.join(HERE, "defaults", "policy.yml"),
 )
+github_arguments.add_argument("--retry-count", type=int, default=240)
```
Comment (felickz) [inline ghascompliance/__main__.py:53]: Would be good to expose via 

https://github.com/advanced-security/policy-as-code/blob/main/action.yml

**Review by felickz — APPROVED** [review]
👍

**Comment by GeekMasher** [comment]
Seems there is a bug, tracking it down now

File: action.yml [hunk 2]
```diff
@@ -35,6 +35,10 @@ inputs:
     description: Policy as Code branch
     default: main
 
+  retries:
+    description: Number of times to retry the action
+    default: 240 # 1 hours worth of retries
```
Comment (felickz) [inline action.yml:40]: 🫶

**Comment by GeekMasher** [comment]
Bug is resolved, ready to merge once confirmed in production

**Pull request merged**